### ISSUE TITLE
Does Plaso support Elastic Cloud?

### ISSUE THREAD

**Issue by GenericUser450** [log2timeline/plaso#3671]
**Description of problem:**

Hiya,

I'm trying to use psort to push a .plaso file into Elastic Cloud, but it's failing with the error below. I was wondering if there's support for Elastic Cloud? If so, what piece am I missing here? If not, when will this be a possibility? 

**Command Line:**

```
psort.py -o elastic --elastic_mappings /etc/plaso.mappings --index_name test_index --server us-test.es.us-east-1.aws.found.io --port 9243 --elastic_user <user> --elastic_password  <password>  win7.plaso
```

**Debug output/tracebacks:**

Traceback:

```
 File "/usr/bin/psort.py", line 101, in <module>
    if not Main():
  File "/usr/bin/psort.py", line 74, in Main
    tool.ProcessStorage()
  File "/usr/lib/python3/dist-packages/plaso/cli/psort_tool.py", line 598, in ProcessStorage
    analysis_engine.ExportEvents(
  File "/usr/lib/python3/dist-packages/plaso/multi_processing/psort.py", line 1049, in ExportEvents
    output_module.WriteHeader()
  File "/usr/lib/python3/dist-packages/plaso/output/elastic.py", line 20, in WriteHeader
    self._CreateIndexIfNotExists(self._index_name, self._mappings)
  File "/usr/lib/python3/dist-packages/plaso/output/shared_elastic.py", line 222, in _CreateIndexIfNotExists
    if not self._client.indices.exists(index_name):
  File "/usr/lib/python3/dist-packages/elasticsearch/client/utils.py", line 152, in _wrapped
    return func(*args, params=params, headers=headers, **kwargs)
  File "/usr/lib/python3/dist-packages/elasticsearch/client/indices.py", line 320, in exists
    return self.transport.perform_request(
  File "/usr/lib/python3/dist-packages/elasticsearch/transport.py", line 392, in perform_request
    raise e
  File "/usr/lib/python3/dist-packages/elasticsearch/transport.py", line 358, in perform_request
    status, headers_response, data = connection.perform_request(
  File "/usr/lib/python3/dist-packages/elasticsearch/connection/http_urllib3.py", line 269, in perform_request
    self._raise_error(response.status, raw_data)
  File "/usr/lib/python3/dist-packages/elasticsearch/connection/base.py", line 300, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(
elasticsearch.exceptions.RequestError: RequestError(400, '')
```

**Comment by joachimmetz** [comment]
We've not tested Plaso on "Elastic Cloud", if it just provides an Elasticsearch 7 instance I guess it should work. Based on your tracebase something going wrong in connection to the Elasticsearch instance, but unfortunately no details in the RequestError

**Comment by GenericUser450** [comment]
It appears to spin up a pretty normal cluster - the elastic port is different from default, but I defined that in the command string so no biggie there.

**ElasticSearch build info:**

>   "version" : {
    "number" : "7.13.0",
    "build_flavor" : "default",
    "build_type" : "docker",
    "build_hash" : "5ca8591c6fcdb1260ce95b08a8e023559635c6f3",
    "build_date" : "2021-05-19T22:22:26.081971330Z",
    "build_snapshot" : false,
    "lucene_version" : "8.8.2",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "You Know, for Search"

**Cluster:**
![image](https://user-images.githubusercontent.com/81480690/120384918-c651f200-c2f4-11eb-8afe-a7cac0682b1e.png)

FYI, Elastic Cloud can be built on Azure, AWS and Google Cloud. You just choose which one you want to use and it deploys the same configuration to each. 

Since the RequestError didn't have additional details it kinda leaves me in a weird spot since I don't specifically know what the issue was with the connection.

**Comment by GenericUser450** [comment]
I'm going to answer my own question as I figured this out. The issue was two-fold. 

1. I was using capital letters in my index name
2. The connection issue was because I didn't explicitly define use-ssl in my command. Once I updated it to the following it worked:

> psort.py -o elastic --elastic_mappings /etc/plaso.mappings --index_name us_test --server us-test.es.us-east-1.aws.found.io --port 9243 --use-ssl --elastic_user <user> --elastic_password <password> win7.plaso

So, I can verify that plaso does work without a hitch with Elastic Cloud as of right now!
